src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java

```diff
@@ -110,7 +113,103 @@ public void putLoanPolicyStorageLoanPoliciesByLoanPolicyId(
     Handler<AsyncResult<Response>> asyncResultHandler,
     Context vertxContext) {
 
-    MyPgUtil.putUpsert204(LOAN_POLICY_TABLE, entity, loanPolicyId, okapiHeaders, vertxContext,
-        PutLoanPolicyStorageLoanPoliciesByLoanPolicyIdResponse.class, asyncResultHandler);
+    validate(entity, asyncResultHandler, () ->
+      MyPgUtil.putUpsert204(LOAN_POLICY_TABLE, entity, loanPolicyId, okapiHeaders, vertxContext,
+          PutLoanPolicyStorageLoanPoliciesByLoanPolicyIdResponse.class, asyncResultHandler));
```

Comment:
Personally, I would prefer that the validate method returned something that allowed the process to continue or report an error, as that would allow the validate method to be organised differently and make the steps clearer. I understand that is challenging to do in this code at the moment.

Alternatively, there could be two parameters, one for what to do when valid and the other for what to do when invalid, which might be a consumer of an error response.